MigrationPreparer/MigrationPreparerImplementation.cpp

```diff
@@ -0,0 +1,595 @@
+            lineNumber[key] = ++curLineIndex;
+            valueEntry.push_back(newValue);
+            dataStore.Close();
+            dataStoreMutex.unlock();
+            return Core::ERROR_NONE;
+        }
+
+        // Handle Update Request
+        if(lineNumber.find(key) != lineNumber.end()) {
+
+            string oldValue = valueEntry[lineNumber[key] - 2];
+
+            if(oldValue == newValue) {
+                LOGWARN("Entry {%s:%s} is already existing in the dataStore, returning success", oldValue.c_str(), key.c_str());
+                dataStoreMutex.unlock();
+                return Core::ERROR_NONE;
+            }
+            
+            // sed command to replace value in the dataStore
+            oldValue = escapeSed(oldValue, PATTERN);
+            string escapedNewValue = escapeSed(newValue, REPLACEMENT);
+            int result = v_secure_system("/bin/sed -i -E '%ss/%s/%s/' %s", std::to_string(lineNumber[key]).c_str(), oldValue.c_str(), escapedNewValue.c_str(), DATASTORE_PATH);
+            
+            if (result != -1 && WIFEXITED(result)) {
+                valueEntry[lineNumber[key] - 2] = newValue;
+                dataStoreMutex.unlock();
+                return Core::ERROR_NONE;
+            }
+            result = WEXITSTATUS(result);
+            LOGERR("Failed to update entry for {%s:%s} in migration datastore, v_secure_system failed with error %d",key.c_str(), newValue.c_str(), result);
+            dataStoreMutex.unlock();
+            return ERROR_WRITE;
+        }
+
+        // Handle subsequent Write request
+        if(!dataStore.Open(false)) {
+            LOGERR("Failed to create migration datastore %s, errno: %d, reason: %s", DATASTORE_PATH, errno, strerror(errno));
+            LOGERR("Failed to create entry for {%s:%s} in migration datastore", key.c_str(), newValue.c_str());
+            dataStoreMutex.unlock();
+            return ERROR_OPEN;
+        }
+
+        // append new key-value pair to the dataStore
+        if(!dataStore.Position(false, dataStore.Size() - 2)) {
+            LOGERR("DataStore truncate failed with errno: %d, reason: %s\n", errno, strerror(errno));
+            LOGERR("Failed to create entry for {%s:%s} in migration datastore", key.c_str(), newValue.c_str());
+            dataStore.Close();
+            dataStoreMutex.unlock();
+            return ERROR_WRITE;
+        }
+        entry = string(",\n \"") + key + string("\":") + newValue + string("\n}"); 
+        dataStore.Write(reinterpret_cast<const uint8_t*>(&entry[0]), entry.size());
+        lineNumber[key] = ++curLineIndex;
+        valueEntry.push_back(newValue);
+        dataStore.Close();
+        dataStoreMutex.unlock();
+        return Core::ERROR_NONE;
+    }
+    
+    uint32_t MigrationPreparerImplementation::readEntry(const string& name, string &result) {
+
+        LOGINFO("[READ] params={name: %s}", name.c_str());
+        string key = name;
+
+        if(!fileExist) { 
```

Comment:
"fileExist" is not a global variable, it's a private member variable used for convenient sake.
Since in places of read or delete api's or in any other places if we have to check if the dataStore file exist., a new object of Core::File type has to be created and Exists() method has to be invoked upon it like below
        WPEFramework::Core::File dataStore(DATASTORE_PATH);
        if(!dataStore.Exists()) { ...
In order to avoid creating a new object just to check if the file exist, created this "fileExist" Boolean variable.